pkg/internal/webhook/validator.go

```diff
@@ -93,38 +92,46 @@ func (v *validator) validate(ctx context.Context, obj runtime.Object) (admission
 		// Sort list so testing is deterministic.
 		sort.Strings(unrecognisedNames)
 		for _, name := range unrecognisedNames {
-			el = append(el, field.NotSupported(fldPath.Child("plugins"), name, v.registeredPlugins))
+			fieldErrs = append(fieldErrs, field.NotSupported(fldPath.Child("plugins"), name, v.registeredPlugins))
 		}
 	}
 
 	if policy.Spec.Selector.IssuerRef == nil && policy.Spec.Selector.Namespace == nil {
-		el = append(el, field.Required(fldPath.Child("selector"), "one of issuerRef or namespace must be defined, hint: `{}` on either matches everything"))
+		fieldErrs = append(fieldErrs, field.Required(fldPath.Child("selector"), "one of issuerRef or namespace must be defined, hint: `{}` on either matches everything"))
 	}
 
 	if nsSel := policy.Spec.Selector.Namespace; nsSel != nil && len(nsSel.MatchLabels) > 0 {
 		if _, err := metav1.LabelSelectorAsSelector(&metav1.LabelSelector{MatchLabels: nsSel.MatchLabels}); err != nil {
-			el = append(el, field.Invalid(fldPath.Child("selector", "namespace", "matchLabels"), nsSel.MatchLabels, err.Error()))
+			fieldErrs = append(fieldErrs, field.Invalid(fldPath.Child("selector", "namespace", "matchLabels"), nsSel.MatchLabels, err.Error()))
 		}
 	}
 
+	allAllowed := true
 	for _, webhook := range v.webhooks {
 		response, err := webhook.Validate(ctx, policy)
 		if err != nil {
 			return nil, err
 		}
 		if !response.Allowed {
-			for _, e := range response.Errors {
-				el = append(el, e)
-			}
-			// do not allow a CertificateRequestPolicy if it was not
-			// allowed by a plugin that did not set any errors
-			// TODO: when webhooks implement Name() method, provide a plugin name here
-			if len(response.Errors) < 1 {
-				el = append(el, errors.New("a plugin did not allow the CertificateRequest for unknown reasons"))
-			}
+			fieldErrs = append(fieldErrs, response.Errors...)
+
+			allAllowed = false
 		}
 		warnings = append(warnings, response.Warnings...)
 	}
 
-	return warnings, el.ToAggregate()
+	var errs []error
+
+	if aggregateError := fieldErrs.ToAggregate(); aggregateError != nil {
```

Comment:
@irbekrm `ToAggregate()` also de-duplicates all the errors, this was missing in the original PR